modules/gdscript/language_server/gdscript_language_protocol.cpp

```diff
@@ -334,13 +334,49 @@ bool GDScriptLanguageProtocol::is_goto_native_symbols_enabled() const {
 	return bool(_EDITOR_GET("network/language_server/show_native_symbols_in_editor"));
 }
 
+#define STR(s) #s
+// clang-format off
+#define SET_DOCUMENT_METHOD(method) set_method(STR(textDocument/method), callable_mp(text_document.ptr(), &GDScriptTextDocument::method))
+#define SET_COMPLETION_METHOD(method) set_method(STR(completionItem/method), callable_mp(text_document.ptr(), &GDScriptTextDocument::method))
+#define SET_WORKSPACE_METHOD(method) set_method(STR(workspace/method), callable_mp(workspace.ptr(), &GDScriptWorkspace::method))
```

Comment:
Transferred CC @Faless

```suggestion
// clang-format off
#define SET_DOCUMENT_METHOD(method) set_method("textDocument/method", callable_mp(text_document.ptr(), &GDScriptTextDocument::method))
#define SET_COMPLETION_METHOD(method) set_method("completionItem/method", callable_mp(text_document.ptr(), &GDScriptTextDocument::method))
#define SET_WORKSPACE_METHOD(method) set_method("workspace/method", callable_mp(workspace.ptr(), &GDScriptWorkspace::method))
```

Reply:
I would also use `m_method` for the parameter, as the `m_` prefix is a convention we usually use for macro parameters.

And with the direct string usage, you don't need to disable clang-format anymore.

Reply:
Hm indeed, it seems macro expansion doesn't affect string literals.

Then my only suggestion would be to use `_STR` from `typedefs.h` instead of redefining an equivalent here.

Reply:
Didn't know we had a macro for this already. Done